### ISSUE TITLE
[bitnami/nginx] new release is failing on read-only filesystem with no TLS config

### ISSUE THREAD

**Issue by baptiste-bonnaudet** [bitnami/containers#30774]
### Name and Version

bitnami/nginx:1.23-debian-11

### What architecture are you using?

amd64

### What steps will reproduce the bug?

Hi,

The recent [[bitnami/nginx] Release 1.23.4-debian-11-r7](https://github.com/bitnami/containers/pull/30704/files#top) introduced a bug on read-only file systems.

It breaks existing installations which do not use the TLS features of nginx (e.g. TLS termination at LB level, or dev http-only setups)

### What is the expected behavior?

TLS cert should not be re-generated by default if not present in `${NGINX_CONF_DIR}/bitnami/certs/server.crt`

### What do you see instead?

`genrsa: Can't open "/opt/bitnami/nginx/conf/bitnami/certs/server.key" for writing, Read-only file system`

**Comment by javsalgar** [comment]
Hi,

Thank you so much for opening the ticket. We removed the certificate because it was being reported by scanners as a vulnerability in all of our web-based applications. In this sense, one option that we could do is to create an env var like `NGINX_GENERATE_SAMPLE_CERTS`, that would skip this part. However, you may require to mount your own certificates or to mount a volume in `/opt/bitnami/nginx/conf/bitnami/certs`. Thoughts?

**Comment by g-vdn** [comment]
Having similar issue using this container in unRaid.  Certs are gone, link inside container is dead, container fails to start due to this. Wanting to use HTTP, but it won't let me.. On my network I have NGINX running as a reverse proxy, so would like to use the certs there.
I removed all HTTPS variables from container, now having only 'NGINX_HTTP_PORT_NUMBER'='8080' and the port map of course , but the https conf file is still used, so will ask for server.crt and key .  Any suggestions?

![image](https://user-images.githubusercontent.com/58138236/232726663-ed6712b9-6030-4817-af84-8366df8e02e5.png)

![image](https://user-images.githubusercontent.com/58138236/232726855-6c33c566-e539-452e-a0a4-b4f054c9efeb.png)

**Comment by mrbusche** [comment]
This issue still exists in `bitnami/nginx:1.23` and `bitnami/nginx:1.24`

**Comment by mrbusche** [comment]
I'm still receiving the same error pointing to `1.23.4-debian-11-r15`, `1.24.0-debian-11-r6`, or either floating tag.

```Dockerfile
# removed some company specific logic
ARG NODE_VERSION=18
ARG NGINX_VERSION=1.24.0-debian-11-r6

FROM node:${NODE_VERSION}-bullseye AS builder

WORKDIR /workspace

RUN mkdir /output
WORKDIR output
RUN echo "hello" > index.html

FROM bitnami/nginx:${NGINX_VERSION}

COPY --from=builder /output /app
```

Container logs

```
Thu, May 11 2023 12:59:21 pm | nginx 17:59:21.20
Thu, May 11 2023 12:59:21 pm | nginx 17:59:21.20 Welcome to the Bitnami nginx container
Thu, May 11 2023 12:59:21 pm | nginx 17:59:21.20 Subscribe to project updates by watching https://github.com/bitnami/containers
Thu, May 11 2023 12:59:21 pm | nginx 17:59:21.20 Submit issues and feature requests at https://github.com/bitnami/containers/issues
Thu, May 11 2023 12:59:21 pm | nginx 17:59:21.30
Thu, May 11 2023 12:59:21 pm | nginx 17:59:21.40 INFO ==> ** Starting NGINX setup **
Thu, May 11 2023 12:59:21 pm | nginx 17:59:21.50 INFO ==> Validating settings in NGINX_* env vars
Thu, May 11 2023 12:59:21 pm | mkdir: cannot create directory '/opt/bitnami/nginx/conf/bitnami/certs': Read-only file system
```

**Comment by mrbusche** [comment]
> Hi @mrbusche,
> 
> In your case, I guess you are also deleting the `/opt/bitnami/nginx/conf/bitnami` in your Dockerfile after the section of code you shared.
> 
> The folder `/opt/bitnami/nginx/conf/bitnami/certs` already exists in the `bitnami/nginx` and is linked to the `/certs` directory.
> 
> The line `ensure_dir_exists $certs_dir` is trying to regenerate the missing folder, causing that error.
> 
> I'm working on a fix to also skip the certificate generation in case the folder does not exist

I am not deleting certs in the Dockerfile, what I included is what I'm running.

**Comment by migruiz4** [comment]
Hi @mrbusche,

Could you please try using `bitnami/nginx:1.24.0-debian-11-r8`? 

We implemented a fix (https://github.com/bitnami/containers/pull/34043) that should resolve your issue.

The same fix will be released for version 1.23 soon.

**Comment by migruiz4** [comment]
Thank you for your feedback @mrbusche and sorry for the inconvenience!

I will now close this issue as resolved, if you find any further issues please let us know.
